lib/ramble/ramble/software_environments.py

```diff
@@ -29,29 +34,18 @@ class SoftwarePackage(object):
     def __init__(
         self,
         name: str,
-        spec: str,
-        compiler: str = None,
-        compiler_spec: str = None,
-        package_manager: str = "spack",
+        pkg_info: dict,
```

Comment:
That sounds like a good idea. To confirm my understanding, is the idea to have a wrapper class around this pkg_info info, so that the fetching is cleaner?